internal/codespaces/grpc/client_test.go

```diff
@@ -0,0 +1,68 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/cli/cli/v2/internal/codespaces/grpc/test"
+)
+
+func TestMain(m *testing.M) {
+	// Start the gRPC server in the background
+	go func() {
+		err := test.StartServer()
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	m.Run()
+}
+
+func connect(t *testing.T) (ctx context.Context, client *Client) {
+	ctx = context.Background()
```

Comment:
You can make this this function a test helper, with `t.Helper()`: https://pkg.go.dev/testing#T.Helper

You can also define a clean up function that way callers don't need to worry about cleaning up which is useful to keep tests clean. For example:

```go
t.Cleanup(func() {
	client.Close()
})
```

Reply:
Didn't know about this, thanks!